test/functional/lua/spellfile_spec.lua

```diff
@@ -0,0 +1,160 @@
+local n = require('test.functional.testnvim')()
+local t = require('test.testutil')
+
+local exec = n.exec
+local exec_lua = n.exec_lua
+local mkdir_p = n.mkdir_p
+local write_file = t.write_file
+local eq = t.eq
+
+describe('nvim.spellfile', function()
+  before_each(function()
+    n.clear()
+  end)
+
+  it('no-op when .spl and .sug already exist on rtp', function()
+    mkdir_p('Xplug/spell')
+    write_file('Xplug/spell/en_gb.utf-8.spl', 'dummy')
+    write_file('Xplug/spell/en_gb.utf-8.sug', 'dummy')
+    exec('set rtp+=' .. 'Xplug')
+
+    local out = exec_lua([[
+    local s = require('nvim.spellfile')
+
+    local my_spell = vim.fs.joinpath(vim.fn.fnamemodify('Xplug', ':p'), 'spell')
+    local old_access = vim.uv.fs_access
+    vim.uv.fs_access = function(p, mode)
+      return p == my_spell
+    end
+
+    local prompted = false
+    vim.fn.input = function() prompted = true; return 'n' end
+
+    local requests = 0
+    local orig_req = vim.net.request
+    vim.net.request = function(...) requests = requests + 1 end
```

Comment:
If the plugin switches to something other than `vim.net.request`, this will will just continue passing. Example of why mocks are really discouraged.